scripts/build.ts

```diff
@@ -0,0 +1,27 @@
+import { spawn } from 'child_process';
+import fs from 'fs';
+
+export async function build(): Promise<void> {
+  // Remove the current dist dir
+  fs.rmSync('dist', { recursive: true, force: true });
+
+  // Build typescript
+  await new Promise(resolve => {
+    const childProcess = spawn('yarn', ['tsc'], {
+      stdio: 'inherit',
+    });
+
+    childProcess.on('close', () => {
+      resolve('');
```

Comment:
The callback for `close` should get the exit code as an argument.  Might be worthwhile to check it an reject if it is non-zero.  So something like:
```
    childProcess.on('close', (code) => {
      if(code != 0) {
        reject('Build failed');
      } else {
        resolve('');
      }
    });
```

Reply:
I thought about this but figured it wasn't really necessary, since we're already directly logging the output from from `tsc`

Reply:
Doesn't this mean that the `build` function will always succeed even `yarn tsc` fails?

Reply:
Oooh. Yep. When the build fails it logs "Command failed with exit code 2" so I assumed we were good, but the actual exit code is zero due to `spawn`. Good catch